docs/src/main/asciidoc/security-authentication-mechanisms.adoc

```diff
@@ -81,6 +81,69 @@ The resulting digest is used as a key for AES-256 encryption of the cookie value
 The cookie contains an expiry time as part of the encrypted value, so all nodes in the cluster must have their clocks synchronized.
 At one-minute intervals, a new cookie gets generated with an updated expiry time if the session is in use.
 
+To get started with form authentication, you should have similar settings as described in xref:security-basic-authentication-howto.adoc[Enable Basic authentication] and property `quarkus.http.auth.form.enabled` must be set to `true`.
+
+Simple application using form-base authentication can look similar to this:
+
+[source,java]
+----
+@Path("")
+public class FormResource {
+
+    @GET
+    @Path("hello")
+    @RolesAllowed("user")
+    public String hello(@Context SecurityContext security) {
+        return "Hello " + security.getUserPrincipal().getName();
+    }
+
+    @GET
+    @Path("error")
+    public String error() {
+        return "Try login again";
+    }
+}
+----
+
+[source,properties]
+----
+quarkus.http.auth.form.enabled=true
+
+quarkus.http.auth.form.login-page=login.html
+quarkus.http.auth.form.landing-page=hello
+quarkus.http.auth.form.error-page=error
+
+# Define testing user
+quarkus.security.users.embedded.enabled=true
+quarkus.security.users.embedded.plain-text=true
+quarkus.security.users.embedded.users.alice=alice 
+quarkus.security.users.embedded.roles.alice=user
+----
+
+[IMPORTANT]
+====
+Configuring user names, secrets, and roles in the application.properties file is appropriate only for testing scenarios. For securing a production application, it is crucial to use a database to store this information.
```

Comment:
LDAP is also an option, it also uses DB indirectly, but may be it is worth saying `database or LDAP` and link to the basic authentication with JPA doc for an example.